lib/Dialect/TTNN/Utils/OptimizerOverrides.cpp

```diff
@@ -189,4 +189,335 @@ void InputLayoutOverrideParser::print(
+  return enableOptimizer;
+}
+
+bool OptimizerOverridesHandler::getMemoryReconfig() const {
+  return enableMemoryReconfig;
+}
+bool OptimizerOverridesHandler::getMemoryLayoutAnalysis() const {
+  return enableMemoryLayoutAnalysis;
+}
+bool OptimizerOverridesHandler::getEnableMemoryLayoutAnalysisPolicy() const {
+  return enableMemoryLayoutAnalysisPolicy;
+}
+MemoryLayoutAnalysisPolicyType
+OptimizerOverridesHandler::getMemoryLayoutAnalysisPolicy() const {
+  return memoryLayoutAnalysisPolicy;
+}
+
+std::string OptimizerOverridesHandler::getSystemDescPath() const {
+  return systemDescPath;
+}
+int64_t OptimizerOverridesHandler::getMaxLegalLayouts() const {
+  return maxLegalLayouts;
+}
+std::vector<int64_t> OptimizerOverridesHandler::getMeshShape() const {
+  return meshShape;
+}
+
+llvm::StringMap<InputLayoutOverrideParams>
+OptimizerOverridesHandler::getInputLayoutOverrides() const {
+  return inputLayoutOverrides;
+}
+llvm::StringMap<OutputLayoutOverrideParams>
+OptimizerOverridesHandler::getOutputLayoutOverrides() const {
+  return outputLayoutOverrides;
+}
+
+std::string OptimizerOverridesHandler::toString() const {
+
+  std::string options = "";
+
+  if (enableOptimizer) {
+    options += "enable-optimizer=true ";
+  }
+
+  if (enableMemoryReconfig) {
+    options += "memreconfig-enabled=true ";
+  }
+
+  if (enableMemoryLayoutAnalysis) {
+    options += "memory-layout-analysis-enabled=true ";
+  }
+
+  if (enableMemoryLayoutAnalysisPolicy) {
+    options += "memory-layout-analysis-policy=";
+    switch (memoryLayoutAnalysisPolicy) {
+    case MemoryLayoutAnalysisPolicyType::DFSharding:
+      options += "DFSharding ";
+      break;
+    case MemoryLayoutAnalysisPolicyType::L1Interleaved:
+      options += "L1Interleaved ";
+      break;
+    }
+  }
+
+  // Create input layout overrides.
+  //  Example: insert-memreconfig=input0=0:1,input1=0,input2=0:1:2
+  if (inputLayoutOverrides.size() > 0) {
+    options += "insert-memreconfig=";
+    // Read operation name and operand indexes.
+    for (const auto &entry : inputLayoutOverrides) {
+      // Concatenate operation name.
+      options += std::string(entry.getKey()) + "=";
+      // Concatenate operand indexes.
+      const InputLayoutOverrideParams &params = entry.getValue();
+      // Concatenate operand indexes.
+      for (int64_t operandIdx : params.operandIdxes) {
+        options += std::to_string(operandIdx) + ":";
+      }
+      // Remove the last colon.
+      options.pop_back();
+      options += ",";
+    }
+    // Remove the last comma.
+    options.pop_back();
+    // Add a space for the next option.
+    options += " ";
+  }
+
+  // Create output layout overrides.
+  //  Example:
+  //  override-output-layout=op1=2x2:dram:interleaved:tile:fp32,op2=4x4:l1:block_sharded:row_major:fp16
+  //  Example:
+  //  override-output-layout=add_1_2=1x1:dram:interleaved:row_major:f32"
+  if (outputLayoutOverrides.size() > 0) {
```

Comment:
Could you reuse OutputLayoutOverrideParser print method? Same applies for inputLayoutOverrides.

Reply:
I think I need to change a little bit the method. I can write for each segment separate method and call it from toString.

Reply:
Done.